benchmark/benchmark_revisions.py

```diff
@@ -0,0 +1,168 @@
+                        subprocess.run(["git", "fetch", "-q"], check=True)
+                        subprocess.run(["git", "checkout", revision, "-q"], check=True)
+            else:
+                try:
+                    # If we already downloaded a revision we don't need to clone the whole
+                    # PL repository. Instead we just copy the first revision in the directory
+                    # checkout will then get the correct revision
+                    first_revision = next((x for x in revisions_directory.iterdir() if x.is_dir()))
+                    print(">>> Revision not found locally, copying...")
+                    shutil.copytree(str(first_revision), str(revisions_directory / revision))
+
+                except StopIteration:
+                    # Didn't find a local revision to copy, use git to clone
+                    print(">>> Revision not found locally, cloning...")
+
+                    pl_directory.mkdir()
+                    with cd(revisions_directory):
+                        subprocess.run(
+                            [
+                                "git",
+                                "clone",
+                                "https://www.github.com/xanaduai/pennylane",
+                                revision,
+                                "-q",
+                            ],
+                            check=True,
+                        )
+
+                with cd(pl_directory):
+                    subprocess.run(["git", "checkout", "master", "-q"], check=True)
+                    subprocess.run(["git", "fetch", "-q"], check=True)
+                    res = subprocess.run(["git", "checkout", revision, "-q"])
+
+                # An error occured during checkout, so the revision does not exist
+                if res.returncode != 0:
+                    print(
+                        col(
+                            ">>> Couldn't check out revision {}, deleting temporary data".format(
+                                revision
+                            ),
+                            "red",
+                        )
+                    )
+
+                    # Regular rmtree hickups with read-only files. We thus use an errorhandler that tries to mark them
+                    # as writeable and retries. See also
+                    # https://stackoverflow.com/questions/1889597/deleting-directory-in-python/1889686
+                    shutil.rmtree(str(pl_directory), onerror=remove_readonly)
+                    continue
+
+            benchmark_file_path = Path(__file__).parent / "benchmark.py"
+            benchmark_env = os.environ.copy()
+            benchmark_env["PYTHONPATH"] = str(pl_directory) + ";" + benchmark_env["PATH"]
+            subprocess.run(
+                ["python3", str(benchmark_file_path)] + unknown_args + ["--noinfo"],
+                env=benchmark_env,
+                check=True,
+            )
+        except:
```

Comment:
Will section ever get with `check=True`? (I'm not too experienced with using `subprocess.run`, just checking